homeassistant/components/knx/websocket.py

```diff
@@ -77,21 +80,92 @@ async def register_panel(hass: HomeAssistant) -> None:
         )
 
 
+type KnxWebSocketCommandHandler = Callable[
+    [HomeAssistant, KNXModule, websocket_api.ActiveConnection, dict[str, Any]], None
+]
+type KnxAsyncWebSocketCommandHandler = Callable[
+    [HomeAssistant, KNXModule, websocket_api.ActiveConnection, dict[str, Any]],
+    Awaitable[None],
+]
+
+
+@overload
+def provide_knx(
+    func: KnxAsyncWebSocketCommandHandler,
+) -> websocket_api.const.AsyncWebSocketCommandHandler: ...
+@overload
+def provide_knx(
+    func: KnxWebSocketCommandHandler,
+) -> websocket_api.const.WebSocketCommandHandler: ...
+
+
+def provide_knx(
+    func: KnxAsyncWebSocketCommandHandler | KnxWebSocketCommandHandler,
+) -> (
+    websocket_api.const.AsyncWebSocketCommandHandler
+    | websocket_api.const.WebSocketCommandHandler
+):
+    """Websocket decorator to provide a KNXModule instance."""
+
+    def _send_not_loaded_error(
+        connection: websocket_api.ActiveConnection, msg_id: int
+    ) -> None:
+        connection.send_error(
+            msg_id,
+            websocket_api.const.ERR_HOME_ASSISTANT_ERROR,
+            "KNX integration not loaded.",
+        )
+
+    if asyncio.iscoroutinefunction(func):
+
+        @wraps(func)
+        async def with_knx(
+            hass: HomeAssistant,
+            connection: websocket_api.ActiveConnection,
+            msg: dict[str, Any],
+        ) -> None:
+            """Add KNX Module to call function."""
+            try:
+                knx: KNXModule = hass.data[DOMAIN]
+            except KeyError:
+                _send_not_loaded_error(connection, msg["id"])
+                return
+            await func(hass, knx, connection, msg)
```

Comment:
`websocket_api.async_response` creates eager tasks. Should this be used here too, instead of `await`?

Reply:
It should already be running in the task at this point AFAICT. I don't think we need to create another one